### PR TITLE
Invert document instead of geometry

### PR DESCRIPTION
**Pull request by tyehle**
If the page size has not been set, then fall back to inverting just the geometry.

Possible solution to #25

### CODE REVIEW

**Comment by tatarize** [comment]
Now suffers a minor conflict with merged corrections for #26

I am inclined to merge this as well, however, I do have to classify fixing something like this as a breaking change. So `0.13.0`

**Comment by tyehle** [comment]
Yeah, I think it would be possible to add a flag in the config (eg: `invert_document` or something) to turn on this behavior, but to me it seems that flipping the document what people would want almost all of the time.

**Comment by abey79** [comment]
I like this and agree that this is what most people want in most cases. It will break 90+% of existing setups though (most physical coord systems are top-left/y-down). May I suggest adding a `vertical_flip` flag (along with a `horizontal_flip` flag for orthogonality). This wording might resonate best when people find y-inverted results on first try. This flag would work along side the existing `invert_y` (probably not the best idea to use both but why not), and would fail with error if `page_size` is undefined.

Advantages:
- not a breaking change
- good/intuitive wording

Drawback:
- some redundancy with existing `invert_y` & friends (could be addressed in the doc by splitting "basic" vs. "advanced/probably don't need" parameters).

File: vpype_gcode/gwrite.py [hunk 1]
```diff
@@ -18,10 +18,20 @@ def invert_axis(document: vp.Document, invert_x: bool, invert_y: bool):
...
     the center of the bounds.
     """
 
-    bounds = document.bounds()
+    # if the page size is set, then we'll flip around the middle of the page,
+    # otherwise around the geometry bounds
+    if document.page_size is not None:
+        # page size is in px, but everything else is in mm
+        (x, y) = document.page_size
+        px_per_mm = vp.utils.UNITS["mm"]
```
Comment (abey79) [inline vpype_gcode/gwrite.py:30]: `vp.UNITS` would be the canonical syntax here (vpype imports everything public in the top-level namespace).

Comment (abey79) [reply vpype_gcode/gwrite.py:30]: Also, this is very likely incorrect. At this stage, the document is converted to whatever unit is set by the config file (see [this line](https://github.com/plottertools/vpype-gcode/blob/595b15bd2fbb90b89349b1cc299875d206d79be3/vpype_gcode/gwrite.py#L115). Hard-coding units here will likely break any setup with a different unit. Some refactoring might be needed to address this.

Comment (tyehle) [reply vpype_gcode/gwrite.py:30]: Ahh, so that is what's going on! Was wondering about things being in mm. Seemed strange.

**Comment by tyehle** [comment]
@abey79 Changed implementation as you suggested, with `horizontal_flip` and `vertical_flip` config options, and fixed the hardcoded units mess by doing the flipping before any scaling.

**Review by tatarize — APPROVED** [review]
I'll wait for abey but this no longer seems breaking. Just option for a different type of flip.

**Comment by abey79** [comment]
I have a small doubt with the order of operation. Prior to this PR, the axis inversion happened after the scaling/offsetting. Now the opposite is true. I haven't checked, but I believe that this will change the offset sign on axes where inversion/flipping happens. This is probably a breaking change.

**Comment by tyehle** [comment]
@abey79 you are right. I changed the order back to what it was before

**Comment by tyehle** [comment]
I think all the concerns about this have now been resolved. Is this good to merge?

**Comment by abey79** [comment]
Yes, this looks good to me!

**Review by abey79 — APPROVED** [review]

**Pull request merged**